exe/Main.hs

```diff
@@ -179,15 +180,26 @@ showEvent lock (EventFileDiagnostics (toNormalizedFilePath -> file) diags) =
 showEvent lock e = withLock lock $ print e
 
 
-cradleToSession :: Cradle a -> IO HscEnvEq
-cradleToSession cradle = do
+loadGhcSessionIO :: FilePath -> Rules ()
+loadGhcSessionIO dir = do
+    defineNoFile $ \(GetHscEnv opts deps) ->
```

Comment:
Why do we need this separate rule? Is it just to get more caching, i.e., you don’t need to reinitialize the session if the cabal file changed but the resulting flags did not change? Would be good to document this.